### DISCUSSION TITLE
(Force) Close on "feerates are too different" while processing UpdateAddHtlc

### DISCUSSION THREAD

**Discussion by DerEwige** [ACINQ/eclair#2577] [Node operators]
I had a strange automated channel close tonight.

First of all it showed as mutual close in the channel explorers.
([mempool.space](https://mempool.space/tx/cbb940893c289a9623ad6e4fee3260466c2d0dbb132b294e0055fd4efb688ff8))

But I believe this is just because all the funds was still on the openers side.

Am I correct in my assumption that this line was the trigger of a force close?

`eclair.2023-01-22_03.log:2023-01-22 03:30:51,101 ERROR f.a.e.c.fsm.Channel PAY n:025022f660d278b8d90415bdde1b385841b6caf179dc69569e4b55fd5aec70555f c:8993449f974a47d6417af229b643bbebfdb23c941e6237c6f435912695da94d8 - local/remote feerates are too different: remoteFeeratePerKw=3260 localFeeratePerKw=253 while processing msg=UpdateAddHtlc in state=NORMAL
`

**Comment by rorp** [comment]
This is a well known issue, just try to search “feerates are too different” here.

The proper solution is using anchor channels. Which is not always possible, especially for legacy channels.

So there’s a workaround. Set `eclair.on-chain-fees.feerate-tolerance.ratio-low` and `eclair.on-chain-fees.feerate-tolerance.ratio-high` as low and high respectively as you feel comfortable.

Here are the default values: https://github.com/ACINQ/eclair/blob/master/eclair-core/src/main/resources/reference.conf#L214

**Reply by DerEwige** [reply]
I'm aware of these settings and I have them at 0.1 and 10

But I thought they were only relevant when doing mutual close.
I did not know they were relevant all the time when not using anchor channels.

The channel was opened from the other side. I will have to check, why no anchor channel was used

**Reply by t-bast** [reply]
It is quite surprising that there's a more than 10x feerate difference between your node and your peer...we know feerate estimation is inherently inaccurate, but it's bad that the gap is so wide (and there isn't much we can do, apart from using anchor channels).
